pkg/pmem-state/pmme-state.go

```diff
@@ -0,0 +1,147 @@
+/*
+Copyright 2019  Intel Corporation.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+package pmemstate
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/pkg/errors"
+	"k8s.io/klog/glog"
+)
+
+// GetAllFunc callback function used for StateManager.GetAll().
+// This function is called with ID, for each entry found in the state.
+type GetAllFunc func(id string) bool
+
+// StateManager manages the driver persistent state, i.e, volumes information
```

Comment:
How does the state manager handle concurrency? This needs to be documented to ensure that either the manager or the caller do the necessary locking.

Reply:
Done. Implemented concurrency. Added tests to test this.